### PR TITLE
scratch3 join flow redirects to /join, outside editor

### PR DESCRIPTION
**Pull request by benjiwheeler**
### Resolves:

Step towards resolving https://github.com/LLK/scratch-www/issues/3053

### Changes:

instead of popping up new join flow in a modal, this makes most pages redirect to /join. Except for editor.

### CODE REVIEW

File: test/unit/components/navigation.test.jsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+const {shallowWithIntl} = require('../../helpers/intl-helpers.jsx');
+import configureStore from 'redux-mock-store';
+const Navigation = require('../../../src/components/navigation/www/navigation.jsx');
+const sessionActions = require('../../../src/redux/session.js');
+
+describe('Navigation', () => {
+    const mockStore = configureStore();
+    let store;
+
+    beforeEach(() => {
```
Comment (picklesrus) [inline test/unit/components/navigation.test.jsx]: should store be reset between tests?

File: test/unit/components/navigation.test.jsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            .dive() // unwrap redux connect(injectIntl(JoinFlow))
+            .dive(); // unwrap injectIntl(JoinFlow)
+    };
+
+    test('when using old join flow, clicking Join Scratch attemps to open registration', () => {
+        store = mockStore({
+            navigation: {
+                useScratch3Registration: false
+            },
+            session: {
```
Comment (picklesrus) [inline test/unit/components/navigation.test.jsx]: Do you need to set session and messageCount in these tests?

File: src/components/navigation/www/navigation.jsx [hunk 3]
```diff
@@ -78,6 +78,13 @@ class Navigation extends React.Component {
         if (!this.props.user) return;
         return `/users/${this.props.user.username}/`;
     }
+    handleClickRegistration (event) {
```
[the remark this answers is not in the document]
Comment (benjiwheeler) [reply src/components/navigation/www/navigation.jsx]: Done. (I simulate a click, which calls this)

File: test/unit/components/navigation.test.jsx (hunk 1 above)
Comment (benjiwheeler) [reply test/unit/components/navigation.test.jsx]: hm, good idea. doing this

File: test/unit/components/navigation.test.jsx (hunk 2 above)
Comment (benjiwheeler) [reply test/unit/components/navigation.test.jsx]: i think we do need session (there's a conditional in render that checks it), but we don't need messageCount (which I think isn't even spelled right! Ack!)

File: test/unit/components/navigation.test.jsx (hunk 1 above)
Comment (benjiwheeler) [reply test/unit/components/navigation.test.jsx]: Done

**Pull request merged**